osv/models.py

```diff
@@ -332,18 +332,25 @@ def _tokenize(self, value):
     value_lower = value.lower()
     return re.split(r'\W+', value_lower) + [value_lower]
 
+  def _normalize(self, name):
+    """Normalize package name per https://peps.python.org/pep-0503/#normalized-names"""
+
+    return re.sub(r"[-_.]+", "-", name).lower()
+
   def _pre_put_hook(self):
     """Pre-put hook for populating search indices."""
     search_indices = set()
 
     search_indices.update(self._tokenize(self.id()))
 
-    for affected_package in self.affected_packages:
+    for pkg in self.affected_packages:
       # Set PURL if it wasn't provided.
-      if not affected_package.package.purl:
-        affected_package.package.purl = purl_helpers.package_to_purl(
+      if not pkg.package.purl:
+        pkg.package.purl = purl_helpers.package_to_purl(
             ecosystems.normalize(affected_package.package.ecosystem),
-            affected_package.package.name)
+            pkg.package.name)
+      if pkg.package.ecosystem and pkg.package.ecosystem == 'PyPI':
```

Comment:
nit: no need for the first part of the if, as if it's not set the second comparison will fail anyway. 

That said, I would suggest refactoring this into its own function like so:

```python
def _normalize_package(package):
  """docstring here"""
  if package.ecosystem == 'PyPI':
     # per link...
    return re.sub(r"[-_.]+", "-", package.name).lower()

#  ... in the pre_put_hook 
for pkg in self.affected_packages:
  pkg.package = _normalize_package(pkg.package)
```

Reply:
Addressed in 47438b76ae7c3b24ee97d85478bc01325d888bb6